packages/ragbits-core/src/ragbits/core/vector_stores/chroma.py

```diff
@@ -80,14 +104,20 @@ async def store(self, entries: list[VectorStoreEntry]) -> None:
         # TODO: Think about better id components for hashing
         ids = [sha256(entry.key.encode("utf-8")).hexdigest() for entry in entries]
         embeddings = [entry.vector for entry in entries]
-        metadatas = [
-            {
-                "__key": entry.key,
-                "__metadata": json.dumps(entry.metadata, default=str),
-            }
-            for entry in entries
-        ]
-        self._collection.add(ids=ids, embeddings=embeddings, metadatas=metadatas)  # type: ignore
+
+        if self.metadata_store is not None:
+            for key, meta in zip(ids, [entry.metadata for entry in entries], strict=False):
+                await self.metadata_store.store(key, meta)
+            metadata_to_store = None
+        else:
+            metadata_to_store = [
+                {self.METADATA_INNER_KEY: json.dumps(entry.metadata, default=str)} for entry in entries
```

Comment:
I think we should put the raw entry metadata here, serializing it to a string makes it useless for per-field queries

Reply:
Created a task for it: https://github.com/deepsense-ai/ragbits/issues/157